src/vivarium_dcpn_hypertension_mgt/tools/utilities.py

```diff
@@ -32,35 +68,68 @@ def prep_external_data(data_file, location):
         data.sex = data.sex.apply(lambda s: s.strip())
         data = data[data.sex == 'Both']
 
+    data = transform_data(data_file.stem, data)
+
     if 'sex' in data and set(data.sex) == {'Both'}:  # duplicate for male, female
         male = data
         male.sex = 'Male'
         female = data.copy()
         female.sex = 'Female'
         data = pd.concat([male, female])
 
-    if data_file.stem in CONFIDENCE_INTERVALS_TO_CONVERT:
-        for k in CONFIDENCE_INTERVALS_TO_CONVERT[data_file.stem]:
-            data = convert_confidence_interval(data, k)
+    return reshape(data)
 
-    if data_file.stem == 'baseline_treatment_profiles':
-        data = collapse_other_drug_profiles(data)
-    return data
 
+def transform_data(data_type, data):
+    spec = TRANSFORMATION_SPECIFICATION[data_type]
+    measure_data = []
+
+    for m in spec['measures']:
+        measure_data.append(create_draw_level_data(data, m, spec['columns']))
+
+    return pd.concat(measure_data)
+
+
+def create_draw_level_data(data, measure, columns_to_keep):
+    data['measure'] = measure
```

Comment:
You're doing a bunch of non-draw sampling data manipulation here that may be worth refactoring.